lib/ansible/module_utils/tacp_ansible/tacp_utils.py

```diff
@@ -9,10 +9,261 @@
+        self.api = tacp.ApplicationsApi(api_client)
+        self.api_client = api_client
+
+    def get_all(self):
+        """ Get all application instances as a list of dicts
+        """
+        api_response = self.api.get_applications_using_get()
+        if api_response:
+            instance_dicts = []
+            for instance in api_response:
+                instance_dicts.append(instance.to_dict())
+            return instance_dicts
+        return None
+
+    def get_uuid_by_name(self, name):
+        """ Return the UUID of an appplication instance if it exists
+            given the name of the instance, None otherwise
+        """
+        assert name is not None
+
+        all_instances = self.get_all()
+        for instance in all_instances:
+            if instance['name'] == name:
+                return instance['uuid']
+        return None
+
+    def get_by_uuid(self, uuid):
+        """ Returns an application as a dict specified by UUID, None otherwise
+        """
+        assert uuid is not None
+
+        api_response = self.api.get_application_using_get(uuid)
+        if api_response:
+            return api_response.to_dict()
+        return None
+
+    def create(self, body):
+        """ Creates an application with the provided API request body, returns
+            bool of success/failure
+        """
+        assert body is not None
+
+        api_response = self.api.create_application_from_template_using_post(
+            body)
+        try:
+            wait_for_action_to_complete(
+                api_response.action_uuid, self.api_client)
+            return api_response.object_uuid
+        except ActionTimedOutException:
+            return "Exception when waiting for action to complete, action timed out."
+        except InvalidActionUuidException:
+            return "Exception when waiting for action to complete, invalid action UUID."
+        except Exception as e:
+            return e
+
+    def delete(self, uuid):
+        """ Deletes an application specified by UUID, returns bool of success/failure
+        """
+        assert uuid is not None
+
+        api_response = self.api.delete_application_using_delete(uuid)
+        try:
+            wait_for_action_to_complete(
+                api_response.action_uuid, self.api_client)
+            return True
+        except ActionTimedOutException:
+            return "Exception when waiting for action to complete, action timed out."
+        except InvalidActionUuidException:
+            return "Exception when waiting for action to complete, invalid action UUID."
+        except Exception as e:
+            return e
+
+    def power_action_on_instance_by_uuid(self, uuid, power_action):
+        """ Performs a specified power operation on an application instance
+            specified by UUID, returns bool of success/failure
+        """
+        power_action_dict = {
+            ACTION_started: self.api.start_application_using_put,
```

Comment:
If we need this mapping somewhere else too, we should centralize it on the `Action` class

```
Action:
    STARTED = 'started'
    # ...
    CHOICES = {
        STARTED: 'start_application_using_get',
        # ...
    }
```

Then you can

```
assert action in Action.CHOICES
resp = getattr(self.api, Action.CHOICES[action])(uuid)
```